Add signers example

lorisleiva opened the pull request.

lorisleiva commented:
* **#3048** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3048?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lorisleiva and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

steveluscher commented on examples/signers/src/example.ts:
```diff
@@ -0,0 +1,197 @@
+        source: signer, // <- We pass the signer here, not just the address.
+    });
+
+    // Get the latest blockhash to use as a transaction lifetime constraint.
+    const rpc = createSolanaRpc('http://127.0.0.1:8899');
+    const { value: latestBlockhash } = await rpc.getLatestBlockhash().send();
```
Ooh, is it not preferable to reduce the complexity of this thing by:

1. Removing the comment about setting up the validator from the top of this file
2. Jamming a fake blockhash in here

You'll still be able to sign stuff, but you won't have to set up a validator or import `createSolanaRpc` to this example.

lorisleiva replied:
Goot shout! I've deleted all requirements for a test validator by using a mock blockhash.

lorisleiva commented:
@steveluscher 

> I feel like the string -> seed -> keypair example might be confusing to some [...] and is sort of orthogonal to just trying to demonstrate the signers API.

Yeah that's a fair point. I've changed this logic to simply use the first 32 bytes of the `example-keypair.json` file. Which brings us to the next point.

> Would this example land harder if the imported signers had the same public key address?

By making the change mentioned above, we know have option 2 and 3 which uses the same signer. I've also changed the no-op signer to use the same address so now option 2, 3 and 4 are the same signer. I believe option 1 cannot be included in this though since we're trying to showcase generated signers.

steveluscher commented on examples/signers/src/example.ts:
```diff
@@ -0,0 +1,197 @@
+ * This polyfill may be required to access the Web Crypto API in your environment.
+ */
+// @ts-expect-error FIXME(https://github.com/solana-labs/solana-web3.js/pull/3047)
+install();
```
That doesn't seem right. Says it was [implemented in `v18.4.0`](https://github.com/WICG/webcrypto-secure-curves/issues/20#issuecomment-1346208580). Let's investigate this.

steveluscher replied:
K, so in Node 18, `crypto` is not injected into the global scope by default. You have to add `--experimental-global-webcrypto` after `tsx`.

You can do this (to all the examples) or not, but I think we're saying that the new web3.js supports Node LTS and up, which at this point is `>20.16.0`.

steveluscher replied:
Either way I'd prefer not to have the polyfill in the example, lest people copy and paste it blindly.

lorisleiva replied:
Thank you for looking into this. I thought we were still supporting Node 18 which is why I was using this version but I've bumped it to 20+ and everything works without the polyfill now so I've removed it. Good to know it still works with 18 with an experimental flag though.

steveluscher reviewed (approved):
![Added via Giphy](https://media4.giphy.com/media/LP7GkiMTFm4gg/giphy.gif?cid=c9487743cnnq8knhzyh133nuez1kwpb6psmz47f4nxjgq9n3&ep=v1_gifs_search&rid=giphy.gif&ct=g)

lorisleiva commented:
### Merge activity

* **Aug 7, 4:34 AM EDT**: @lorisleiva started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3048).
